### PR TITLE
fdroidserver: use ruamel.yaml 0.17.21

### PR DESCRIPTION
**Pull request by linsui**
The output is changed in the newer versions. Using a different version make the output of `fdroid rewritemeta` different with upstream.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Atemu — COMMENTED** [review]
Could you separate the actual change and the refactoring into two commits?

**Comment by linsui** [comment]
Done.

**Review by Atemu — APPROVED** [review]
Diff LGTM and the software still runs successfully.

Is there an easy way to verify whether this fixes the issue it intends to fix?

**Comment by Atemu** [comment]
(Side note: If you actively use this software it'd be great if you could add yourself to the maintainers :)

**Comment by linsui** [comment]
> Is there an easy way to verify whether this fixes the issue it intends to fix?

You can download https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.looker.droidify.yml to `./metadata/com.looker.droidify.yml` directory. Then run `fdroid rewritemeta com.looker.droidify`. Before this change the metadata file will be changed.

By the way, F-Droid always use fdroidserver from master so this still doesn't always output the same file as upstream. I'm not sure if I should update fdroidserver to master.

**Review by Atemu — APPROVED** [review]
I could repro the issue and can confirm it's fixed.

I'll give @obfusk a day to review aswell though.

**Comment by Atemu** [comment]
I'm personally of the opinion that packages should be at the version they're most useful at. If the only valid use-case is to check upstream f-droid (who use master), then we might want this tool to use something fresher than the latest release.

I can't speak for others on this matter though.

**Comment by linsui** [comment]
Per discussion with @obfusk and @jugendhacker, I removed @obfusk from the maintainer list and add me and @jugendhacker . See https://matrix.to/#/!UAdCANfoQdxlMXhNuM:f-droid.org/$iX28GZF34GLNev45AryeSHS5l1reOzYBWZTaK8hHnBI?via=f-droid.org&via=matrix.org

> I don't work on fdroidserver or nix any more. so you should probably remove me as maintainer.

and https://matrix.to/#/!UAdCANfoQdxlMXhNuM:f-droid.org/$_QXIC-KMgD-_o1stTNzlBmMo-fTJW4rffYyXmkYdEJk?via=f-droid.org&via=matrix.org

> if you want a second maintainer for the fdroidserver nix package you could add me, if you want

**Review by jugendhacker — APPROVED** [review]
Looks good to me

File: pkgs/development/tools/fdroidserver/default.nix [hunk 1]
```diff
@@ -58,7 +78,13 @@ python.pkgs.buildPythonApplication rec {
     pyyaml
     qrcode
     requests
-    ruamel-yaml
+    (ruamel-yaml.overrideAttrs (old: {
```
Comment (dotlambda) [inline pkgs/development/tools/fdroidserver/default.nix:81]: This is not allowed. Put it in `overrideScope` instead.

**Pull request merged**